core/shared/src/main/scala/zio/metrics/MetricKey.scala

```diff
@@ -26,12 +26,27 @@ import zio.stacktracer.TracingImplicits.disableAutoTrace
  * boundaries of a histogram. In this way, it is impossible to ever create
  * different metrics with conflicting keys.
  */
-final case class MetricKey[+Type] private (
+sealed case class MetricKey[+Type] private (
   name: String,
   keyType: Type,
   tags: Set[MetricLabel] = Set.empty
 ) { self =>
 
+  def description: Option[String] = None
+
+  def copy[Type2](name: String = name, keyType: Type2 = keyType, tags: Set[MetricLabel] = tags): MetricKey[Type2] =
+    new MetricKey(name, keyType, tags) {
+      override def description: Option[String] = self.description
+    }
+
+  /**
+   * Returns a new `MetricKey` with the specified description.
+   */
+  def described(description0: String): MetricKey[Type] =
```

Comment:
We should delete this and instead provide variants of the constructors allowing specifying a description just like we did for the metrics themselves since there should be no way to update the description once a metric is created.

Reply:
@adamgfraser sure, done!